onnx/defs/math/defs.cc

```diff
@@ -2594,4 +2594,157 @@ ONNX_OPERATOR_SET_SCHEMA(
             propagateShapeFromInputToOutput(ctx, 0, 1);
           }
         }));
+
+static int64_t fft_signal_ndim_default = 1;
+
+static const char* FFT_ver13_doc = R"DOC(
+Fast Fourier Transform.
+
+This implementation suppose the DFT (Discrete Fourier Transform) defined by
+```
+A_k =  \sum_{m=0}^{n-1} a_m \exp\left\{-2\pi i{mk \over n}\right\}
+\qquad k = 0,\ldots,n-1.
+```
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    FFT,
+    13,
+    OpSchema()
+        .SetDoc(FFT_ver13_doc)
+        .Attr(
+            "signal_ndim",
+            "The number of dimension of the input signal."
+            "Values can be 1, 2 or 3.",
+            AttributeProto::INT,
+            fft_signal_ndim_default)
+        .Input(
+            0,
+            "input",
+            "A complex signal of dimension signal_ndim."
+            "The last dimension of the tensor should be 2,"
+            "representing the real and imaginary components of complex numbers,"
+            "and should have at least signal_ndim + 2 dimensions."
+            "The first dimension is the batch dimension.",
+            "T")
+        .Output(
+            0,
+            "output",
+            "The fourier transform of the input vector,"
+            "using the same format as the input.",
+            "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)", "tensor(double)"},
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+            propagateElemTypeFromInputToOutput(ctx, 0, 0);
+            const int signal_ndim = getAttribute(ctx, "signal_ndim", fft_signal_ndim_default);
+        
+            if (!hasInputShape(ctx, 0)) {
```

Comment:
Statically unknown shapes are allowed during inference. Inference should not fail in these cases, but can leave the output shapes unspecified.